File: t-tasks/src/main/java/com/teo/ttasks/ui/fragments/task_lists/TaskListsView.java
package com.teo.ttasks.ui.fragments.task_lists;

import com.teo.ttasks.ui.base.MvpView;
import com.teo.ttasks.ui.items.TaskListItem;

import java.util.List;

interface TaskListsView extends MvpView {

    void onTaskListsLoading();

    void onTaskListsEmpty();

    void onTaskListsError();

    void onTaskListsLoaded(List<TaskListItem> taskListItems);

    void onTaskListUpdateError();
}

File: t-tasks/src/main/java/com/teo/ttasks/ui/fragments/task_lists/TaskListsView.kt
package com.teo.ttasks.ui.fragments.task_lists

import com.teo.ttasks.ui.base.MvpView
import com.teo.ttasks.ui.items.TaskListItem

internal interface TaskListsView : MvpView {

    fun onTaskListsLoading()

    fun onTaskListsEmpty()

    fun onTaskListsError()

    fun onTaskListsLoaded(taskListItems: List<TaskListItem>)

    fun onTaskListUpdateError()
}
